user_management/src/user/views/user_id.py

```diff
@@ -17,3 +19,17 @@ def get(request, user_id):
             return JsonResponse(data={'id': user.id, 'username': user.username}, status=200)
         except Exception as e:
             return JsonResponse(data={'errors': [f'An unexpected error occurred : {e}']}, status=500)
+
+
+class UserIdListView(View):
+    @staticmethod
+    def post(request):
+        id_list = json.loads(request.body).get('id_list')
+        if not isinstance(id_list, list):
+            return JsonResponse(data={'errors': ['id_list should be a list']}, status=400)
+        response = []
+        for user_id in id_list:
+            user = User.objects.filter(id=user_id).first()
+            if user is not None:
+                response.append({'id': user.id, 'username': user.username})
+        return JsonResponse(data=response, status=200, safe=False)
```

Comment:
No need to spam the db:

```py
@staticmethod
    def post(request):
        try:
            id_list = json.loads(request.body).get('id_list')
            if not isinstance(id_list, list):
                return JsonResponse(data={'errors': ['id_list should be a list']}, status=400)

            users = User.objects.filter(id__in=id_list)
            response = [{'id': user.id, 'username': user.username} for user in users]
            return JsonResponse(data=response, status=200, safe=False)

        except Exception as e:
            return JsonResponse(data={'errors': [f'An unexpected error occurred : {e}']}, status=500)
```

Also could be better if the response was a hashmap (key=id, value=username) instead of an array